### PR TITLE
Update gitignore faq with correct steps

### PR DESCRIPTION
**Pull request by mirnawong1**
Updating FAQ with correct steps per [loom](https://www.loom.com/share/9b3b8e2b617f41a8bad76ec7e42dd014) recorded by @will-sargent-dbtlabs 

Also added faq as src in other related documents.

### CODE REVIEW

**Review by will-sargent-dbtlabs — APPROVED** [review]
@mirnawong1 Reviewed and tweaked.  I suspect you will probably be able to make it even more visually interesting with my additions.  I can re-review it if you like, after you take a look. Thanks!

**Comment by mirnawong1** [comment]
thank you so much @will-sargent-dbtlabs , this is wonderful! i'm going to dedicate reviewing this tomorrow (fri) because I've prioritized semantic layer and ide sqlfluff docs today but on it!!

File: website/docs/docs/collaborate/git/version-control-basics.md [hunk 1]
```diff
@@ -57,6 +57,25 @@ Refer to [merge conflicts](/docs/collaborate/git/merge-conflicts) to learn how t
 
 ## The .gitignore file
 
-dbt Labs recommends that you exclude files so they're not tracked by Git and won't slow down your dbt project.
+For proper operation, dbt Cloud requires that certain sub-folders in the git repository that contains your dbt project be excluded from tracking by git.
```
Comment (mirnawong1) [inline website/docs/docs/collaborate/git/version-control-basics.md]:
```suggestion
To make sure dbt Cloud runs smoothly, you must exclude certain sub-folders in your git repository containing your dbt project from being tracked by git.
```

File: website/docs/docs/collaborate/git/version-control-basics.md [hunk 2]
```diff
@@ -57,6 +57,25 @@ Refer to [merge conflicts](/docs/collaborate/git/merge-conflicts) to learn how t
 
 ## The .gitignore file
 
-dbt Labs recommends that you exclude files so they're not tracked by Git and won't slow down your dbt project.
+To make sure dbt Cloud runs smoothly, you must exclude certain sub-folders in your git repository containing your dbt project from being tracked by git.
 
-You can do this with a special file named [.gitignore](https://github.com/dbt-labs/dbt-starter-project/blob/main/.gitignore) which is automatically included in your dbt project after you initialize it in dbt Cloud. The `.gitignore` file must be placed at the root of your dbt project.
+This is achieved by adding three lines to a special file named [.gitignore](https://github.com/dbt-labs/dbt-starter-project/blob/main/.gitignore). This file is placed in the root folder of your dbt project.
+
+Some git providers will auto-create a 'boilerplate' `.gitignore` file when the repository is created.  However, in dbt Labs' experience, none of these boilerplate `.gitignore` files will contain the entries necessary for the proper operation of dbt Cloud.
```
Comment (mirnawong1) [inline website/docs/docs/collaborate/git/version-control-basics.md]:
```suggestion
Some git providers will auto-create a 'boilerplate' `.gitignore` file when the repository is created.  However, based on dbt Labs' experience, these default `.gitignore` files typically don't include the required entries for dbt Cloud to function correctly.
```

File: website/docs/faqs/Git/gitignore.md [hunk 3]
```diff
@@ -1,25 +1,224 @@
...
+> NOTE: The 'main' branch might also be called 'master', 'dev', 'qa', 'prod' or something else depending on the organizational naming conventions.  The goal is to merge these changes into the root branch that all other development branches are created from.
+10. Return to the dbt Cloud IDE.
+11. Use the 'Change Branch' button, to switch to the main branch of the project.
+12. Once the branch has changed, click the **Pull from remote** button to pull in all the changes. You can verify the changes by making sure the files/folders in the .gitignore file are in italics. 
+
+<Lightbox src="/img/docs/dbt-cloud/cloud-ide/gitignore-italics.jpg" width="50%" title="A dbt project on the main branch that has properly configured gitignore folders (highlighted in italics)."/>
+
+### Steps to use the git providers web interface to fix a broken .gitignore:
+   
+Sometimes it is necessary to use the git providers web interface to fix a broken `.gitignore` file. While the exact sequence of clicks will be different for each provider, the overall steps are the same.
```
Comment (mirnawong1) [inline website/docs/faqs/Git/gitignore.md]: hey @will-sargent-dbtlabs, why would it be necessary to use the git providers ui to fix the file?

**Comment by mirnawong1** [comment]
hey @will-sargent-dbtlabs awesome additions! thanks so much and i tweaked it a bit. i wanted to double check if there was any difference in the versioning for the gitignore file that's under the git folder? it doesn't look like there's a difference? 

Also is there a reason why the contents under the 'git' folder are different from the contents under the 'troubleshooting' folder?

Also also (sorry!) shouldn't the changes you added include the tmp folder deletion from project.yml? 

anyway, let me know what you think! it's looking so detailed and i think/hope it'll help users bc i bet they get stuck on this a lot

**Comment by mirnawong1** [comment]
hey @will-sargent-dbtlabs, can you give me a rereview whenever you have a chance? 🙏

File: website/docs/docs/collaborate/git/version-control-basics.md [hunk 4]
```diff
@@ -57,6 +57,26 @@ Refer to [merge conflicts](/docs/collaborate/git/merge-conflicts) to learn how t
...
+```
+dbt_packages/
+logs/
+target/
+```
+
+**Note** &mdash; By using a trailing slash, these lines in the `gitignore` file serve as 'folder wildcards', excluding all files and folders within those folders from being tracked by git. 
+
+
+:::tip
```
Comment (mirnawong1) [inline website/docs/docs/collaborate/git/version-control-basics.md]:
```suggestion
:::note
```

**Pull request merged**